client/src/components/library/LibraryCreateButton.jsx

```diff
@@ -1,18 +1,30 @@
 import React, { Component, PropTypes } from 'react';
+import DashSelect from '../common/DashSelect';
 
 export default class LibraryCreateButton extends Component {
   render() {
     const { onCreate } = this.props;
+    const options = [
+      { value: 'dataset', label: 'Dataset' },
+      { value: 'visualisation', label: 'Visualisation' },
+      { value: 'dashboard', label: 'Dashboard' },
+      { value: 'collection', label: 'Collection' },
+    ];
+    const onChange = (value) => {
+      onCreate(value);
+      // Force a rerender so menu still shows placeholder
+      this.forceUpdate();
```

Comment:
note-to-self/investigate @jonase: It seems that we need to `forceUpdate()` because react-select doesn't do controlled components.